ansible/roles/stig/tasks/TOSS_04_010020.yml

```diff
@@ -10,7 +10,29 @@
 
 
 - name: TOSS-04-010020 - TOSS, for PKI-based authentication, must enforce authorized access to the corresponding private key.
-  # TODO: Implement the task
+  block:
+    - name: Find all private ssh key files
+      ansible.builtin.command: 'find / -xdev -type f -regex ".*/id_[a-z0-9]*[^\.pub]$"'
+      changed_when: false
+      check_mode: false
+      register: nouser_files
+      failed_when: "nouser_files.rc not in [0,1]"
+    - name: Check if the ssh key files have passphrases set
+      ansible.builtin.expect:
+        command: 'ssh-keygen -y -f {{ item }}'
+        responses:
+          'Enter Passphrase:': '*.'
+      register: passphrase_prompts
+      changed_when: false
+      failed_when: false
+      loop: '{{ nouser_files.stdout_lines }}'
+    - name: Verify all the ssh keys prompted for a passphrase
+      ansible.builtin.assert:
+        that:
+          "'Enter passphrase' in item.stdout"
+        success_msg: "The private ssh key has a passphrase"
+        fail_msg: "A private ssh key is does not have a passphrase set. Set it with ssh-keygen -n [passphrase]"
```

Comment:
Is there any way to report which ssh key(s) doesn't have the passphrase ?

Reply:
Right now this test runs for each individual key, so you will get a block of details(including the file path) followed by the success/fail message for each key, so you can tell which ones need to be fixed.